src/vit-servicing-station/vit-servicing-station-tests/src/tests/snapshot/raw_snapshot.rs

```diff
@@ -0,0 +1,220 @@
+use std::time::Duration;
+
+use crate::{
+    common::{
+        clients::RawRestClient,
+        raw_snapshot::{RawSnapshot, RawSnapshotBuilder, RawSnapshotExtension, RawSnapshotUpdater},
+        snapshot::{SnapshotBuilder, VotingPower},
+        startup::quick_start,
+    },
+    tests::snapshot::verifier::assert_raw_against_full_snapshot,
+};
+use assert_fs::TempDir;
+
+use super::verifier::assert_is_empty_raw_against_full_snapshot;
+
+#[test]
+pub fn import_new_raw_snapshot() {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, data) = quick_start(&temp_dir).unwrap();
+    let rest_client = server.rest_client_with_token(&data.token_hash());
+
+    let raw_snapshot = RawSnapshot::default();
+
+    rest_client.put_raw_snapshot(&raw_snapshot).unwrap();
+
+    assert_eq!(
+        vec![raw_snapshot.tag.to_string()],
+        rest_client.snapshot_tags().unwrap(),
+        "expected tags vs tags taken from REST API"
+    );
+
+    let assigner = raw_snapshot.clone().into();
+
+    let snapshot_infos = raw_snapshot
+        .clone()
+        .into_full_snapshot_infos(&assigner)
+        .unwrap();
+
+    for snapshot_info in snapshot_infos.iter() {
+        assert_raw_against_full_snapshot(snapshot_info, raw_snapshot.clone(), &rest_client);
+    }
+}
+
+#[test]
+pub fn reimport_with_empty_raw_snapshot() {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, data) = quick_start(&temp_dir).unwrap();
+    let rest_client = server.rest_client_with_token(&data.token_hash());
+
+    let raw_snapshot = RawSnapshot::default();
+
+    rest_client.put_raw_snapshot(&raw_snapshot).unwrap();
+
+    let empty_snapshot = RawSnapshot::empty(raw_snapshot.tag.clone());
+
+    rest_client.put_raw_snapshot(&empty_snapshot).unwrap();
+
+    let assigner = raw_snapshot.clone().into();
```

Comment:
Aha , i see potential to implement yet another into trait and simplify 
```
let assigner = raw_snapshot.clone().into();
 let snapshot_infos = raw_snapshot
        .clone()
        .into_full_snapshot_infos(&assigner)
        .unwrap();
```

into 

```
  let snapshot_infos = raw_snapshot.clone().try_into() .unwrap();
```

We just need to combine those two intos into one. What you think ?
```
impl TryInto<FullSnapshotInfo> for RawSnapshot {
  type Error = ...
  
  fn try_into(raw: RawSnapshot) -> Result<FullSnapshotInfo,...> {
    let assigner = raw.clone().into();
     raw.into_full_snapshot_infos(&assigner)
  }
}
```

or evem conver into_full_snapshot_info to above impl